core/src/text/textBuffer.h

```diff
@@ -19,31 +14,54 @@ class FontContext;
 class TextBuffer : public LabelMesh {
 
 public:
-
-    TextBuffer(std::shared_ptr<VertexLayout> _vertexLayout);
-    ~TextBuffer();
-
-    /* Create and add TextLabel */
-    bool addLabel(const TextStyle::Parameters& _params, Label::Transform _transform,
-                  Label::Type _type, FontContext& _fontContext);
-
     struct WordBreak {
         int start;
         int end;
     };
 
-    static std::vector<WordBreak> findWords(const std::string& _text);
+    class Builder {
+        std::vector<Label::Vertex> m_scratchVertices;
+        std::vector<Label::Vertex> m_vertices;
+        std::vector<std::unique_ptr<Label>> m_labels;
 
-    virtual void draw(ShaderProgram& _shader) override;
+        glm::vec2 m_bbox;
+        glm::vec2 m_quadsLocalOrigin;
+        int m_numLines;
+        FontContext::FontMetrics m_metrics;
+
+        std::unique_ptr<TextBuffer> m_mesh;
+
+        std::vector<WordBreak> m_wordBreaks;
+
+        int applyWordWrapping(std::vector<FONSquad>& _quads, const TextStyle::Parameters& _params,
+                              const FontContext::FontMetrics& _metrics);
+
+        static std::string applyTextTransform(const TextStyle::Parameters& _params, const std::string& _string);
 
-private:
-    static int applyWordWrapping(std::vector<FONSquad>& _quads, const TextStyle::Parameters& _params,
-                                 const FontContext::FontMetrics& _metrics, Label::Type _type,
-                                 std::vector<TextBuffer::WordBreak>& words);
+    public:
+        void setup(std::shared_ptr<VertexLayout> _vertexLayout);
+        bool prepareLabel(FontContext& _fontContext, const TextStyle::Parameters& _params);
+        void addLabel(const TextStyle::Parameters& _params, Label::Type _type,
+                      Label::Transform _transform);
 
-    static std::string applyTextTransform(const TextStyle::Parameters& _params, const std::string& _string);
+        std::unique_ptr<TextBuffer> build();
+
+        float labelWidth() { return m_bbox.x; }
+
+        static void findWords(const std::string& _text, std::vector<WordBreak>& _words);
+
+    };
+
+    TextBuffer(std::shared_ptr<VertexLayout> _vertexLayout);
+    ~TextBuffer();
+
+    virtual void draw(ShaderProgram& _shader) override;
+
+protected:
 
     bool m_strokePass = false;
+
+    friend class Builder;
```

Comment:
I am not sure there is anything in textBuffer requiring Builder to be a friend. Is this a leftover, or am I missing something?